crates/kitsune_p2p/dht_arc/src/dht_arc_set.rs

```diff
@@ -72,17 +73,19 @@ impl DhtArcSet {
         }
     }
 
-    pub fn from_interval<A: Borrow<ArcInterval>>(arc: A) -> Self {
+    pub fn from_interval<A: Borrow<DhtArc>>(arc: A) -> Self {
         match arc.borrow() {
-            ArcInterval::Full => Self::new_full(),
-            ArcInterval::Empty => Self::new_empty(),
-            ArcInterval::Bounded(start, end) => Self::from_bounds(*start, *end),
+            DhtArc::Full(_) => Self::new_full(),
+            DhtArc::Empty(_) => Self::new_empty(),
+            DhtArc::Bounded(start, end) => Self::from_bounds(*start, *end),
         }
     }
 
-    pub fn intervals(&self) -> Vec<ArcInterval> {
+    pub fn intervals(&self) -> Vec<DhtArc> {
         match self {
-            Self::Full => vec![ArcInterval::Full],
+            // XXX: loss of information here, this DhtArc
+            //      does not actually know about its true start
+            Self::Full => vec![DhtArc::Full(0u32.into())],
```

Comment:
should we even allow this then? or is this test-only and can be `#[cfg(feature = "test_utils")]`?

Reply:
@neonphog I am torn about this.

On the one hand, the original `DhtArc` made sure to always keep track of its center point, even in the full and empty cases where, for the purposes of marking out a section of the circle, it doesn't matter. So I wanted to ensure we still have that.

On the other hand, we do have cases where we really don't care about the centerpoint anymore, and want to lose that information. DhtArcSet is such a case, by the time we are talking about sets of arcs, we are just interested in the intersections and unions of them, meaning we are sometimes joining and splitting arcs anyway and can't keep track of their original origin locations.

I'm not sure if we actually even have cases where we have to be aware of the original location that a DhtArc came from, but I wanted to include the ability in case we do. But then that leads this situation, where a location needs to be specified even when we don't have one.

This distinction is part of why I created `ArcInterval` in the first place, as a way to describe an arc where we don't care about the centerpoint. Now those concerns are collapsed again.

One solution would be to remove this info from DhtArc if we don't actually need it. I'll do a check on this.

@freesig you wrote the original `DhtArc` so I wonder if you have context on why we might need to maintain the centerpoint (now left edges) of the arcs even if they're full or empty?